dp-cluster-setup-utility.py

```diff
@@ -1790,14 +1894,30 @@ def execute(self):
     if not CMPrerequisites(cm).satisfied():
       return 1
     
-    clusters_resp_from_dp = dp.check_cm(cm)
+    #
+    # If number of clusters managed by CM = 1 , The script will execute knox based flow
+    #
+    knox = None
+    clusters_to_register = []
+    clusters_resp_from_dp = []
+    if cm.total_clusters == 1:
+      active_cluster  = cm.clusters[0]
+      knox = KnoxAdminApi(user.url_input('Knox Admin URL', 'knox_admin.url', default=str(active_cluster.knox_url())), user.credential_input('Knox Admin User', 'knox_admin.user'))
+      topology_util = CMTopologyUtil(cm, [])
```

Comment:
Raising here, because it is not available elsewhere. The base class TopologyUtil takes the parameter name as 'ambari'. I would rename that to 'cluster_manager.'

Reply:
actually we can remove it. We do that